build-image/webserver.bash

```diff
@@ -24,7 +24,7 @@ fi
 
 if [[ $1 == "inst_done" ]]; then
   mkdir -p "${TMPDIR:-/tmp}"/webserver
-  sed 's|%HOSTNAME|'"${HOSTNAME:-openhabian}"'|g' /opt/openhabian/includes/webserver/install-complete.html > "${TMPDIR:-/tmp}"/webserver/index.html
+  sed 's|%HOSTNAME|'"${hostname:-openhabian}"'|g' /opt/openhabian/includes/webserver/install-complete.html > "${TMPDIR:-/tmp}"/webserver/index.html
```

Comment:
I believe this should stay as it was before because it will otherwise try to broadcast to an hostname that is not currently its own. this only run during the first boot setup anyway.

Reply:
$hostname is taken from openhabian.conf, sourced at the beginning of first-boot.bash so it should be available here.

Not sure but isn't it the other way round that *if* user changed it (remember: not interactively but in openhabian.conf) ?
Then at this stage the old code ($HOSTNAME) has been wrong ever, hasn't it ? So that was a bug that has gone unspotted so far.

Reply:
But it does work, I know it does because I have used it on every install I have performed. Maybe this just boils down to me being cautious about changing things that work now.

Reply:
> I have used it on every install I have performed

you mean you changed hostname via openhabian.conf *and* watched the webserver output on each of these ?
That's the minimum conditions for the bug to show (if there was/is, as I said not sure).

Reply:
how to proceed, is this the only requested change left? Saw my response?

Reply:
No, I have changed the hostname and the output worked correctly on the original default hostname of openhabian.

Reply:
Err I don't understand that sentence. What code do you refer to, old or new? And what does that have to do with the default hostname ? Do you mean you tried changing it but it rather used the default ("openhabian") ?

Reply:
Nvm, I think I misread. It should be fine. I don't think it will break anything and if it does we can always revert and try again.